tests/sentry/api/serializers/rest_framework/test_list.py

```diff
@@ -0,0 +1,103 @@
+from __future__ import absolute_import
+from sentry.api.serializers.rest_framework import ListField
+
+from rest_framework import serializers
+
+from sentry.testutils import TestCase
+
+
+class ListFieldTest(TestCase):
+    def assert_success(self, serializer, value):
+        assert serializer.is_valid()
+        assert not serializer.errors
+        assert serializer.object == value
+
+    def assert_unsuccessful(self, serializer, errors):
+        assert not serializer.is_valid()
+        assert serializer.errors == errors
+        assert serializer.object is None
+
+    def test_simple(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': [1, 2, 3]})
+        self.assert_success(serializer, {'list_field': [1, 2, 3]})
+
+    def test_single_element_list(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': [1]})
+        self.assert_success(serializer, {'list_field': [1]})
+
+    def test_empty_list_child_required(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(required=False, child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': []})
+        self.assert_success(serializer, {'list_field': []})
+
+    def test_empty_list_child_required_default_value(self):
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(default=[], required=False, child=serializers.IntegerField())
+        serializer = DummySerializer(data={'list_field': []})
+        self.assert_success(serializer, {'list_field': []})
+
+    def test_empty_list_child_required_complex_object(self):
+        class DummyChildSerializer(serializers.Serializer):
+            name = serializers.CharField()
+            age = serializers.IntegerField()
+
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(required=False, child=DummyChildSerializer())
+        serializer = DummySerializer(data={'list_field': []})
+        self.assert_unsuccessful(serializer,
+                                 {'list_field': ['non_field_errors: No input provided']})
+
+    def test_empty_list_child_not_required_complex_object(self):
+        class DummyChildSerializer(serializers.Serializer):
+            name = serializers.CharField()
+            age = serializers.IntegerField()
+
+        class DummySerializer(serializers.Serializer):
+            list_field = ListField(required=False, child=DummyChildSerializer(required=False))
+        serializer = DummySerializer(data={'list_field': []})
+        self.assert_unsuccessful(serializer,
```

Comment:
Also note that even if the child has been set to `required=False` `serializer.is_valid()` is still `False` for more complex serializers. This is because we are giving an empty array.